server/src/main/java/org/apache/iotdb/db/engine/storagegroup/timeindex/TimeIndexLevel.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.engine.storagegroup.timeindex;
+
+public enum TimeIndexLevel {
+  FILE_TIME_INDEX, DEVICE_TIME_INDEX;
+
+  public ITimeIndex getTimeIndex() {
+    switch (this) {
+      case FILE_TIME_INDEX:
+        return new FileTimeIndex();
+      case DEVICE_TIME_INDEX:
+      default:
+        return new DeviceTimeIndex();
```

Comment:
Although FILE_TIME_INDEX is provided, ITimeIndex is heavily bound to devices, as many of its methods involve devices, which leaves FileTimeIndex in an embarrassing position. Maybe it would be better to do some abstraction between DeviceTimeIndex and ITimeIndex and put device-related methods into this abstraction.

Reply:
I'll try this in next PR when supporting the conversion of two kinds of TimeIndex.